lib/kafo/puppet_module.rb

```diff
@@ -49,9 +53,15 @@ def enable
     end
 
     def parse(builder_klass = ParamBuilder)
-      @params      = []
-      @raw_data    = @parser_cache.get(identifier, manifest_path) if @parser_cache
-      @raw_data  ||= @parser.parse(manifest_path)
+      @raw_data = @parser_cache.get(identifier, manifest_path) if @parser_cache
+      if @raw_data.nil?
+        if @parser.nil? || @parser == :none
+          raise ParserError.new("No Puppet module parser is installed and no cache of the file #{manifest_path} is available. Please check debug logs and install optional dependencies for the parser.")
+        else
+          @raw_data = @parser.parse(manifest_path)
```

Comment:
During testing I realized that it would be great if it logged out a debug message, which `@parser.class` which is being used. Maybe a better place would be `self.find_parser` so it would be printed once rather than for every module.